lib/codegen/Decision.cpp

```diff
@@ -148,6 +148,28 @@ void DecisionNode::computeLiveness(std::unordered_set<LeafNode *> &leaves) {
   }
 }
 
+std::vector<DecisionNode *> DecisionNode::topologicalSort(void) {
+  std::vector<DecisionNode *> result;
+  std::vector<DecisionNode *> workList;
+  workList.push_back(this);
+  while (!workList.empty()) {
+    DecisionNode *node = workList.back();
+    workList.pop_back();
+    result.push_back(node);
+    auto it = node->successors2.begin();
```

Comment:
Why use `it` in the loop below and not just `node->successors2.begin()`? Because it's shorter?

Reply:
As I recall I have observed experimentally that it is faster to reuse the iterator that is returned by `erase` rather than to call `begin` again, when you are iterating through a set and deleting every element as you reach it. Moreover, this is essentially a simplification of the more general pattern of looping through a data structure and possibly deleting elements as you go, where you need an iterator because it will not necessarily always point to the beginning of the collection, as it does in this case because we are deleting every element.

Thus I view this code as more idiomatic and faster than repeatedly calling `begin`. Not to mention, the iterator is computed by the `erase` method anyway so it's duplicating work not to use it.